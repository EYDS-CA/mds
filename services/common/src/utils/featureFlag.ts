import { detectProdEnvironment as IN_PROD } from "./environmentUtils";
import flagsmith from "flagsmith";

// Name of feature flags. These correspond to feature flags defined in flagsmith.
export enum Feature {
  MAJOR_PROJECT_ARCHIVE_FILE = "major_project_archive_file",
  DOCUMENTS_REPLACE_FILE = "major_project_replace_file",
  MAJOR_PROJECT_ALL_DOCUMENTS = "major_project_all_documents",
  MAJOR_PROJECT_DECISION_PACKAGE = "major_project_decision_package",
  FLAGSMITH = "flagsmith",
  TSF_V2 = "tsf_v2",
}

// Definition of legacy flags
// TODO: Remove after flagsmith is live
const Flags = {
  [Feature.MAJOR_PROJECT_ARCHIVE_FILE]: !IN_PROD(),
  [Feature.DOCUMENTS_REPLACE_FILE]: !IN_PROD(),
  [Feature.MAJOR_PROJECT_ALL_DOCUMENTS]: !IN_PROD(),
  [Feature.MAJOR_PROJECT_DECISION_PACKAGE]: !IN_PROD(),
  [Feature.TSF_V2]: !IN_PROD(),
  [Feature.FLAGSMITH]: !IN_PROD(),
};

export const initializeFlagsmith = async (flagsmithUrl, flagsmithKey) => {
  await flagsmith.init({
    api: flagsmithUrl,
    environmentID: flagsmithKey,
    cacheFlags: true,
    enableAnalytics: true,
  });
};

/**
 * Returns true if the given feature is enabled
 * @param feature Feature to verify
 * @returns true if the given feature is enabled
 */
export const isFeatureEnabled = (feature: Feature) => {
  if (Feature.FLAGSMITH in Flags && Flags[Feature.FLAGSMITH]) {
    return flagsmith.hasFeature(feature);
  }

  if (feature in Flags) {
    return Flags[feature];
  }

  return false;
};
