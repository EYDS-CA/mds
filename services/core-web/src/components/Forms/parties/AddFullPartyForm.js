import React from "react";
import PropTypes from "prop-types";
import { Field, getFormValues } from "redux-form";
import { Col, Row, Radio } from "antd";
import {
  required,
  email,
  phoneNumber,
  maxLength,
  number,
  postalCodeWithCountry,
} from "@mds/common/redux/utils/Validate";
import { normalizePhone, upperCase, normalizeExt } from "@common/utils/helpers";
import * as FORM from "@/constants/forms";
import CustomPropTypes from "@/customPropTypes";
import { renderConfig } from "@/components/common/config";
import FormWrapper from "@mds/common/components/forms/FormWrapper";
import { useSelector } from "react-redux";

const propTypes = {
  isPerson: PropTypes.bool.isRequired,
  togglePartyChange: PropTypes.func.isRequired,
  provinceOptions: PropTypes.arrayOf(CustomPropTypes.dropdownListItem).isRequired,
};

export const AddFullPartyForm = (props) => {
  const { sub_division_code } = useSelector(getFormValues(FORM.ADD_FULL_PARTY)) ?? {};
  const province = props.provinceOptions.find((prov) => prov.value === sub_division_code);
  const address_type_code = province?.subType;
  return (
    <div>
      <FormWrapper
        onSubmit={() => { }}
        name={FORM.ADD_FULL_PARTY}
        reduxFormConfig={{ destroyOnUnmount: false }}
        isModal
      >
        <Row gutter={48}>
          <Col md={12} sm={24} className="border--right--layout">
            <div className="center margin-large">
              <Radio.Group
                defaultValue
                value={props.isPerson}
                size="large"
                onChange={props.togglePartyChange}
              >
                <Radio.Button value>Person</Radio.Button>
                <Radio.Button value={false}>Organization</Radio.Button>
              </Radio.Group>
            </div>
            <Row gutter={16}>
              <Col md={12} xs={24}>
                <h5>Basic Details</h5>
              </Col>
            </Row>
            {props.isPerson && (
              <Row gutter={16}>
                <Col md={12} xs={24}>
                  <Field
                    id="first_name"
                    name="first_name"
                    label="First Name"
                    component={renderConfig.FIELD}
                    required
                    validate={[required, maxLength(100)]}
                  />
                </Col>
                <Col md={12} xs={24}>
                  <Field
                    id="party_name"
                    name="party_name"
                    label="Surname"
                    component={renderConfig.FIELD}
                    required
                    validate={[required, maxLength(100)]}
                  />
                </Col>
              </Row>
            )}
            {!props.isPerson && (
              <Row gutter={16}>
                <Col span={24}>
                  <Field
                    id="party_name"
                    name="party_name"
                    label="Organization Name"
                    component={renderConfig.FIELD}
                    required
                    validate={[required]}
                  />
                </Col>
              </Row>
            )}
            <Row gutter={16}>
              <Col span={24}>
                <Field
                  id="email"
                  name="email"
                  label="Primary Email"
                  component={renderConfig.FIELD}
                  validate={[email]}
                />
              </Col>
            </Row>
            <Row gutter={16}>
              <Col span={24}>
                <Field
                  id="email_sec"
                  name="email_sec"
                  label="Secondary Email"
                  component={renderConfig.FIELD}
                  validate={[email]}
                />
              </Col>
            </Row>
            <Row gutter={16}>
              <Col span={18}>
                <Field
                  id="phone_no"
                  name="phone_no"
                  label="Primary Phone No."
                  placeholder="e.g. xxx-xxx-xxxx"
                  component={renderConfig.FIELD}
                  required
                  validate={[required, phoneNumber, maxLength(12)]}
                  normalize={normalizePhone}
                />
              </Col>
              <Col span={6}>
                <Field
                  id="phone_ext"
                  name="phone_ext"
                  label="Ext"
                  component={renderConfig.FIELD}
                  validate={[number, maxLength(6)]}
                  normalize={normalizeExt}
                />
              </Col>
            </Row>
            <Row gutter={16}>
              <Col span={18}>
                <Field
                  id="phone_no_sec"
                  name="phone_no_sec"
                  label="Secondary Phone No."
                  placeholder="e.g. xxx-xxx-xxxx"
                  component={renderConfig.FIELD}
                  validate={[phoneNumber, maxLength(12)]}
                  normalize={normalizePhone}
                />
              </Col>
              <Col span={6}>
                <Field
                  id="phone_sec_ext"
                  name="phone_sec_ext"
                  label="Ext"
                  component={renderConfig.FIELD}
                  validate={[number, maxLength(6)]}
                  normalize={normalizeExt}
                />
              </Col>
            </Row>
            <Row gutter={16}>
              <Col span={18}>
                <Field
                  id="phone_no_ter"
                  name="phone_no_ter"
                  label="Tertiary Phone No."
                  placeholder="e.g. xxx-xxx-xxxx"
                  component={renderConfig.FIELD}
                  validate={[phoneNumber, maxLength(12)]}
                  normalize={normalizePhone}
                />
              </Col>
              <Col span={6}>
                <Field
                  id="phone_ter_ext"
                  name="phone_ter_ext"
                  label="Ext"
                  component={renderConfig.FIELD}
                  validate={[number, maxLength(6)]}
                  normalize={normalizeExt}
                />
              </Col>
            </Row>
          </Col>
          <Col md={12} sm={24} style={{ marginTop: "80px" }}>
            <Row gutter={16}>
              <Col md={12} xs={24}>
                <h5>Address</h5>
              </Col>
            </Row>
            <Row gutter={16}>
              <Col span={6}>
                <Field
                  id="suite_no"
                  name="suite_no"
                  label="Suite No."
                  component={renderConfig.FIELD}
                  validate={[maxLength(10)]}
                />
              </Col>
              <Col span={18}>
                <Field
                  id="address_line_1"
                  name="address_line_1"
                  label="Street Address 1"
                  component={renderConfig.FIELD}
                />
              </Col>
            </Row>
            <Row gutter={16}>
              <Col span={18}>
                <Field
                  id="address_line_2"
                  name="address_line_2"
                  label="Street Address 2"
                  component={renderConfig.FIELD}
                />
              </Col>
              <Col span={6}>
                <Field
                  id="sub_division_code"
                  name="sub_division_code"
                  label="Province"
                  component={renderConfig.SELECT}
                  data={props.provinceOptions}
                />
              </Col>
            </Row>
            <Row gutter={16}>
              <Col md={12} xs={24}>
                <Field
                  id="city"
                  name="city"
                  label="City"
                  component={renderConfig.FIELD}
                  validate={[maxLength(30)]}
                />
              </Col>
              <Col md={12} xs={24}>
                <Field
                  id="post_code"
                  name="post_code"
                  label="Postal Code"
                  placeholder="e.g xxxxxx"
                  component={renderConfig.FIELD}
                  validate={[maxLength(10), postalCodeWithCountry(address_type_code)]}
                  normalize={upperCase}
                />
              </Col>
            </Row>
          </Col>
        </Row>
      </FormWrapper>
    </div>
  )
};

AddFullPartyForm.propTypes = propTypes;

export default AddFullPartyForm;
