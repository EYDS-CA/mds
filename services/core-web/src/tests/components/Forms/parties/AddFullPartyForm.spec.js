import React from "react";
import { render } from "@testing-library/react";
import { AddFullPartyForm } from "@/components/Forms/parties/AddFullPartyForm";
import * as MOCK from "@mds/common/tests/mocks/dataMocks";
import { ReduxWrapper } from "@mds/common/tests/utils/ReduxWrapper";


describe("AddFullPartyForm", () => {
  it("renders properly", () => {
    const { container } = render(<ReduxWrapper><AddFullPartyForm
      togglePartyChange={jest.fn()}
      isPerson={false}
      provinceOptions={MOCK.DROPDOWN_PROVINCE_OPTIONS}
    /></ReduxWrapper>);
    expect(container).toMatchSnapshot();
  });
});
