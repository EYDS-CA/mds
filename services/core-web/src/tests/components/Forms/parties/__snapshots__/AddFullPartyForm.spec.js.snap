// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFullPartyForm renders properly 1`] = `
<div>
  <div>
    <form
      class="ant-form ant-form-vertical common-form common-form-ADD_FULL_PARTY form-edit"
      id="ADD_FULL_PARTY"
    >
      <div
        class="ant-row"
        style="margin-left: -24px; margin-right: -24px;"
      >
        <div
          class="ant-col border--right--layout ant-col-sm-24 ant-col-md-12"
          style="padding-left: 24px; padding-right: 24px;"
        >
          <div
            class="center margin-large"
          >
            <div
              class="ant-radio-group ant-radio-group-outline ant-radio-group-large"
            >
              <label
                class="ant-radio-button-wrapper"
              >
                <span
                  class="ant-radio-button"
                >
                  <input
                    class="ant-radio-button-input"
                    type="radio"
                    value="true"
                  />
                  <span
                    class="ant-radio-button-inner"
                  />
                </span>
                <span>
                  Person
                </span>
              </label>
              <label
                class="ant-radio-button-wrapper ant-radio-button-wrapper-checked"
              >
                <span
                  class="ant-radio-button ant-radio-button-checked"
                >
                  <input
                    checked=""
                    class="ant-radio-button-input"
                    type="radio"
                    value="false"
                  />
                  <span
                    class="ant-radio-button-inner"
                  />
                </span>
                <span>
                  Organization
                </span>
              </label>
            </div>
          </div>
          <div
            class="ant-row"
            style="margin-left: -8px; margin-right: -8px;"
          >
            <div
              class="ant-col ant-col-xs-24 ant-col-md-12"
              style="padding-left: 8px; padding-right: 8px;"
            >
              <h5>
                Basic Details
              </h5>
            </div>
          </div>
          <div
            class="ant-row"
            style="margin-left: -8px; margin-right: -8px;"
          >
            <div
              class="ant-col ant-col-24"
              style="padding-left: 8px; padding-right: 8px;"
            >
              <div
                class="ant-form-item ant-form-item-with-help"
              >
                <div
                  class="ant-row ant-form-item-row"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class="ant-form-item-required"
                      for="ADD_FULL_PARTY_party_name"
                      title=""
                    >
                      <div
                        style="width: 100%;"
                      >
                        Organization Name
                      </div>
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-label="party_name"
                          class="ant-input"
                          id="party_name"
                          name="party_name"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      style="display: flex; flex-wrap: nowrap;"
                    >
                      <div
                        class="ant-form-item-explain ant-form-item-explain-connected"
                        id="ADD_FULL_PARTY_party_name_help"
                        role="alert"
                      >
                        <div
                          class=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row"
            style="margin-left: -8px; margin-right: -8px;"
          >
            <div
              class="ant-col ant-col-24"
              style="padding-left: 8px; padding-right: 8px;"
            >
              <div
                class="ant-form-item ant-form-item-with-help"
              >
                <div
                  class="ant-row ant-form-item-row"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class=""
                      for="ADD_FULL_PARTY_email"
                      title=""
                    >
                      <div>
                        Primary Email
                         
                        <span
                          class="form-item-optional"
                        >
                           (optional)
                        </span>
                      </div>
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-label="email"
                          class="ant-input"
                          id="email"
                          name="email"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      style="display: flex; flex-wrap: nowrap;"
                    >
                      <div
                        class="ant-form-item-explain ant-form-item-explain-connected"
                        id="ADD_FULL_PARTY_email_help"
                        role="alert"
                      >
                        <div
                          class=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row"
            style="margin-left: -8px; margin-right: -8px;"
          >
            <div
              class="ant-col ant-col-24"
              style="padding-left: 8px; padding-right: 8px;"
            >
              <div
                class="ant-form-item ant-form-item-with-help"
              >
                <div
                  class="ant-row ant-form-item-row"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class=""
                      for="ADD_FULL_PARTY_email_sec"
                      title=""
                    >
                      <div>
                        Secondary Email
                         
                        <span
                          class="form-item-optional"
                        >
                           (optional)
                        </span>
                      </div>
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-label="email_sec"
                          class="ant-input"
                          id="email_sec"
                          name="email_sec"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      style="display: flex; flex-wrap: nowrap;"
                    >
                      <div
                        class="ant-form-item-explain ant-form-item-explain-connected"
                        id="ADD_FULL_PARTY_email_sec_help"
                        role="alert"
                      >
                        <div
                          class=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row"
            style="margin-left: -8px; margin-right: -8px;"
          >
            <div
              class="ant-col ant-col-18"
              style="padding-left: 8px; padding-right: 8px;"
            >
              <div
                class="ant-form-item ant-form-item-with-help"
              >
                <div
                  class="ant-row ant-form-item-row"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class="ant-form-item-required"
                      for="ADD_FULL_PARTY_phone_no"
                      title=""
                    >
                      <div
                        style="width: 100%;"
                      >
                        Primary Phone No.
                      </div>
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-label="phone_no"
                          class="ant-input"
                          id="phone_no"
                          name="phone_no"
                          placeholder="e.g. xxx-xxx-xxxx"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      style="display: flex; flex-wrap: nowrap;"
                    >
                      <div
                        class="ant-form-item-explain ant-form-item-explain-connected"
                        id="ADD_FULL_PARTY_phone_no_help"
                        role="alert"
                      >
                        <div
                          class=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-col ant-col-6"
              style="padding-left: 8px; padding-right: 8px;"
            >
              <div
                class="ant-form-item ant-form-item-with-help"
              >
                <div
                  class="ant-row ant-form-item-row"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class=""
                      for="ADD_FULL_PARTY_phone_ext"
                      title=""
                    >
                      <div>
                        Ext
                         
                        <span
                          class="form-item-optional"
                        >
                           (optional)
                        </span>
                      </div>
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-label="phone_ext"
                          class="ant-input"
                          id="phone_ext"
                          name="phone_ext"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      style="display: flex; flex-wrap: nowrap;"
                    >
                      <div
                        class="ant-form-item-explain ant-form-item-explain-connected"
                        id="ADD_FULL_PARTY_phone_ext_help"
                        role="alert"
                      >
                        <div
                          class=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row"
            style="margin-left: -8px; margin-right: -8px;"
          >
            <div
              class="ant-col ant-col-18"
              style="padding-left: 8px; padding-right: 8px;"
            >
              <div
                class="ant-form-item ant-form-item-with-help"
              >
                <div
                  class="ant-row ant-form-item-row"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class=""
                      for="ADD_FULL_PARTY_phone_no_sec"
                      title=""
                    >
                      <div>
                        Secondary Phone No.
                         
                        <span
                          class="form-item-optional"
                        >
                           (optional)
                        </span>
                      </div>
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-label="phone_no_sec"
                          class="ant-input"
                          id="phone_no_sec"
                          name="phone_no_sec"
                          placeholder="e.g. xxx-xxx-xxxx"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      style="display: flex; flex-wrap: nowrap;"
                    >
                      <div
                        class="ant-form-item-explain ant-form-item-explain-connected"
                        id="ADD_FULL_PARTY_phone_no_sec_help"
                        role="alert"
                      >
                        <div
                          class=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-col ant-col-6"
              style="padding-left: 8px; padding-right: 8px;"
            >
              <div
                class="ant-form-item ant-form-item-with-help"
              >
                <div
                  class="ant-row ant-form-item-row"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class=""
                      for="ADD_FULL_PARTY_phone_sec_ext"
                      title=""
                    >
                      <div>
                        Ext
                         
                        <span
                          class="form-item-optional"
                        >
                           (optional)
                        </span>
                      </div>
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-label="phone_sec_ext"
                          class="ant-input"
                          id="phone_sec_ext"
                          name="phone_sec_ext"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      style="display: flex; flex-wrap: nowrap;"
                    >
                      <div
                        class="ant-form-item-explain ant-form-item-explain-connected"
                        id="ADD_FULL_PARTY_phone_sec_ext_help"
                        role="alert"
                      >
                        <div
                          class=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row"
            style="margin-left: -8px; margin-right: -8px;"
          >
            <div
              class="ant-col ant-col-18"
              style="padding-left: 8px; padding-right: 8px;"
            >
              <div
                class="ant-form-item ant-form-item-with-help"
              >
                <div
                  class="ant-row ant-form-item-row"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class=""
                      for="ADD_FULL_PARTY_phone_no_ter"
                      title=""
                    >
                      <div>
                        Tertiary Phone No.
                         
                        <span
                          class="form-item-optional"
                        >
                           (optional)
                        </span>
                      </div>
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-label="phone_no_ter"
                          class="ant-input"
                          id="phone_no_ter"
                          name="phone_no_ter"
                          placeholder="e.g. xxx-xxx-xxxx"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      style="display: flex; flex-wrap: nowrap;"
                    >
                      <div
                        class="ant-form-item-explain ant-form-item-explain-connected"
                        id="ADD_FULL_PARTY_phone_no_ter_help"
                        role="alert"
                      >
                        <div
                          class=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-col ant-col-6"
              style="padding-left: 8px; padding-right: 8px;"
            >
              <div
                class="ant-form-item ant-form-item-with-help"
              >
                <div
                  class="ant-row ant-form-item-row"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class=""
                      for="ADD_FULL_PARTY_phone_ter_ext"
                      title=""
                    >
                      <div>
                        Ext
                         
                        <span
                          class="form-item-optional"
                        >
                           (optional)
                        </span>
                      </div>
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-label="phone_ter_ext"
                          class="ant-input"
                          id="phone_ter_ext"
                          name="phone_ter_ext"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      style="display: flex; flex-wrap: nowrap;"
                    >
                      <div
                        class="ant-form-item-explain ant-form-item-explain-connected"
                        id="ADD_FULL_PARTY_phone_ter_ext_help"
                        role="alert"
                      >
                        <div
                          class=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-col ant-col-sm-24 ant-col-md-12"
          style="padding-left: 24px; padding-right: 24px; margin-top: 80px;"
        >
          <div
            class="ant-row"
            style="margin-left: -8px; margin-right: -8px;"
          >
            <div
              class="ant-col ant-col-xs-24 ant-col-md-12"
              style="padding-left: 8px; padding-right: 8px;"
            >
              <h5>
                Address
              </h5>
            </div>
          </div>
          <div
            class="ant-row"
            style="margin-left: -8px; margin-right: -8px;"
          >
            <div
              class="ant-col ant-col-6"
              style="padding-left: 8px; padding-right: 8px;"
            >
              <div
                class="ant-form-item ant-form-item-with-help"
              >
                <div
                  class="ant-row ant-form-item-row"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class=""
                      for="ADD_FULL_PARTY_suite_no"
                      title=""
                    >
                      <div>
                        Suite No.
                         
                        <span
                          class="form-item-optional"
                        >
                           (optional)
                        </span>
                      </div>
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-label="suite_no"
                          class="ant-input"
                          id="suite_no"
                          name="suite_no"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      style="display: flex; flex-wrap: nowrap;"
                    >
                      <div
                        class="ant-form-item-explain ant-form-item-explain-connected"
                        id="ADD_FULL_PARTY_suite_no_help"
                        role="alert"
                      >
                        <div
                          class=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-col ant-col-18"
              style="padding-left: 8px; padding-right: 8px;"
            >
              <div
                class="ant-form-item ant-form-item-with-help"
              >
                <div
                  class="ant-row ant-form-item-row"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class=""
                      for="ADD_FULL_PARTY_address_line_1"
                      title=""
                    >
                      <div>
                        Street Address 1
                         
                        <span
                          class="form-item-optional"
                        >
                           (optional)
                        </span>
                      </div>
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-label="address_line_1"
                          class="ant-input"
                          id="address_line_1"
                          name="address_line_1"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      style="display: flex; flex-wrap: nowrap;"
                    >
                      <div
                        class="ant-form-item-explain ant-form-item-explain-connected"
                        id="ADD_FULL_PARTY_address_line_1_help"
                        role="alert"
                      >
                        <div
                          class=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row"
            style="margin-left: -8px; margin-right: -8px;"
          >
            <div
              class="ant-col ant-col-18"
              style="padding-left: 8px; padding-right: 8px;"
            >
              <div
                class="ant-form-item ant-form-item-with-help"
              >
                <div
                  class="ant-row ant-form-item-row"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class=""
                      for="ADD_FULL_PARTY_address_line_2"
                      title=""
                    >
                      <div>
                        Street Address 2
                         
                        <span
                          class="form-item-optional"
                        >
                           (optional)
                        </span>
                      </div>
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-label="address_line_2"
                          class="ant-input"
                          id="address_line_2"
                          name="address_line_2"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      style="display: flex; flex-wrap: nowrap;"
                    >
                      <div
                        class="ant-form-item-explain ant-form-item-explain-connected"
                        id="ADD_FULL_PARTY_address_line_2_help"
                        role="alert"
                      >
                        <div
                          class=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-col ant-col-6"
              style="padding-left: 8px; padding-right: 8px;"
            >
              <div
                class="ant-form-item ant-form-item-with-help"
              >
                <div
                  class="ant-row ant-form-item-row"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class=""
                      for="ADD_FULL_PARTY_sub_division_code"
                      title=""
                    >
                      <div>
                        Province
                         
                        <span
                          class="form-item-optional"
                        >
                           (optional)
                        </span>
                      </div>
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <div
                          class="ant-select ant-select-in-form-item ant-select-single ant-select-allow-clear ant-select-show-arrow ant-select-show-search"
                          data-cy="sub_division_code"
                        >
                          <div
                            class="ant-select-selector"
                          >
                            <span
                              class="ant-select-selection-search"
                            >
                              <input
                                aria-activedescendant="sub_division_code_list_0"
                                aria-autocomplete="list"
                                aria-controls="sub_division_code_list"
                                aria-haspopup="listbox"
                                aria-owns="sub_division_code_list"
                                autocomplete="off"
                                class="ant-select-selection-search-input"
                                id="sub_division_code"
                                role="combobox"
                                type="search"
                                value=""
                              />
                            </span>
                            <span
                              class="ant-select-selection-placeholder"
                            >
                              Please select
                            </span>
                          </div>
                          <span
                            aria-hidden="true"
                            class="ant-select-arrow"
                            style="user-select: none;"
                            unselectable="on"
                          >
                            <span
                              aria-label="down"
                              class="anticon anticon-down ant-select-suffix"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                />
                              </svg>
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                    <div
                      style="display: flex; flex-wrap: nowrap;"
                    >
                      <div
                        class="ant-form-item-explain ant-form-item-explain-connected"
                        id="ADD_FULL_PARTY_sub_division_code_help"
                        role="alert"
                      >
                        <div
                          class=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row"
            style="margin-left: -8px; margin-right: -8px;"
          >
            <div
              class="ant-col ant-col-xs-24 ant-col-md-12"
              style="padding-left: 8px; padding-right: 8px;"
            >
              <div
                class="ant-form-item ant-form-item-with-help"
              >
                <div
                  class="ant-row ant-form-item-row"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class=""
                      for="ADD_FULL_PARTY_city"
                      title=""
                    >
                      <div>
                        City
                         
                        <span
                          class="form-item-optional"
                        >
                           (optional)
                        </span>
                      </div>
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-label="city"
                          class="ant-input"
                          id="city"
                          name="city"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      style="display: flex; flex-wrap: nowrap;"
                    >
                      <div
                        class="ant-form-item-explain ant-form-item-explain-connected"
                        id="ADD_FULL_PARTY_city_help"
                        role="alert"
                      >
                        <div
                          class=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-col ant-col-xs-24 ant-col-md-12"
              style="padding-left: 8px; padding-right: 8px;"
            >
              <div
                class="ant-form-item ant-form-item-with-help"
              >
                <div
                  class="ant-row ant-form-item-row"
                >
                  <div
                    class="ant-col ant-form-item-label"
                  >
                    <label
                      class=""
                      for="ADD_FULL_PARTY_post_code"
                      title=""
                    >
                      <div>
                        Postal Code
                         
                        <span
                          class="form-item-optional"
                        >
                           (optional)
                        </span>
                      </div>
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-label="post_code"
                          class="ant-input"
                          id="post_code"
                          name="post_code"
                          placeholder="e.g xxxxxx"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      style="display: flex; flex-wrap: nowrap;"
                    >
                      <div
                        class="ant-form-item-explain ant-form-item-explain-connected"
                        id="ADD_FULL_PARTY_post_code_help"
                        role="alert"
                      >
                        <div
                          class=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
